zerver/lib/cache.py

```diff
@@ -173,6 +174,12 @@ def func_with_caching(*args: ParamT.args, **kwargs: ParamT.kwargs) -> ReturnT:
                 return val[0]
 
             val = func(*args, **kwargs)
+            if isinstance(val, QuerySet):
```

Comment:
mypy, somewhat understandably, doesn't approve of this line:

```
zerver/lib/cache.py:177: error: Parameterized generics cannot be used with class or instance checks  [misc]
```

Is there a better way to do this?

Reply:
This is https://github.com/typeddjango/django-stubs/issues/704. I think using a `type: ignore[misc]` should be fine here, which is similar to what we did in #23175. We will be reminded to revisit this whenever it gets fixed upstream.

We can finish up https://github.com/typeddjango/django-stubs/pull/794 after this gets merged.

Reply:
Great, marked it as such!